components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1686,6 +1690,13 @@ private void handleRequestObject(OAuthMessage oAuthMessage, OAuthAuthzRequest oa
              */
         overrideAuthzParameters(oAuthMessage, parameters, oauthRequest.getParam(REQUEST),
                 oauthRequest.getParam(REQUEST_URI), requestObject);
+
+        // If the redirect uri was not given in auth request the registered redirect uri will be available here,
+        // so validating if the registered redirect uri is a single uri that can be properly redirected.
+        if (parameters.getRedirectURI().startsWith(REGEX_PATTERN)) {
```

Comment:
Can we guarantee parameters.getRedirectURI() won't be nulll?

Reply:
RedirectURI is populated in `populateOauthParameters()` method. If it is null, it will be captured here itself.